### PR TITLE
[Perf] Improve PostGIS performance by removing CTEs

### PR DESCRIPTION
**Pull request by jerluc**
This small changeset dramatically improves the performance of several classes of SQL queries by using a simple sub-query for `tableExpression` rather than a CTE.

### CODE REVIEW

**Review by madhulikajc — APPROVED** [review]

**Comment by madhulikajc** [comment]
Looks great, thanks for including me! Should the comment in line 49 change?

**Review by manossaratsis — APPROVED** [review]
✌️

File: postgis_source.go [hunk 1]
```diff
@@ -53,21 +54,20 @@ func (c *PostGISConfig) Dataset() (*goqu.SelectDataset, error) {
 		return nil, InvalidTableConfig
```
Comment (ccma14) [inline postgis_source.go:54]: Nit: Might be nice to also change the comment in L49 (can't annotate there) to no longer talk about "CTE-based SelectDataset"

**Review by ccma14 — APPROVED** [review]

**Review by audreyyku — APPROVED** [review]

File: postgis_source.go (hunk 1 above)
Comment (jerluc) [reply postgis_source.go:54]: Good call (I believe this is what @madhulikajc was referring to as well). I'll update the comment!

**Pull request merged**